src/middleware/log_request.rs

```diff
@@ -60,6 +65,104 @@ pub fn add_custom_metadata<V: Display>(req: &mut dyn RequestExt, key: &'static s
     }
 }
 
+fn report_to_sentry(req: &dyn RequestExt, res: &AfterResult, response_time: u64) {
+    let (message, level) = match res {
+        Err(e) => (e.to_string(), Level::Error),
+        Ok(_) => {
+            if response_time <= SLOW_REQUEST_THRESHOLD_MS {
+                return;
+            }
+
+            let message = format!("Slow Request: {} {}", req.method(), req.path());
+            (message, Level::Info)
+        }
+    };
+
+    let config = |scope: &mut sentry::Scope| {
+        let method = Some(req.method().as_str().to_owned());
+
+        let scheme = match req.scheme() {
+            Scheme::Http => "http",
+            Scheme::Https => "https",
+        };
+
+        let host = match req.host() {
+            Host::Name(name) => name.to_owned(),
+            Host::Socket(addr) => addr.to_string(),
+        };
+
+        let path = &req.extensions().find::<OriginalPath>().unwrap().0;
+
+        let url = format!("{}://{}{}", scheme, host, path).parse().ok();
+
+        {
+            let ip_addr = match req.headers().get("x-real-ip") {
+                Some(value) => value
+                    .to_str()
+                    .ok()
+                    .and_then(|str| str.parse::<IpAddr>().ok()),
+                None => Some(req.remote_addr().ip()),
+            };
+
+            let user = sentry::User {
+                ip_address: ip_addr.map(IpAddress::Exact),
+                ..Default::default()
+            };
+
+            scope.set_user(Some(user));
+        }
```

Comment:
We should not send the IP over to Sentry. Maybe let's set the id and username when the user is logged in?

Reply:
okay. I just added it because it's currently being logged to the console too, but that should be easy to remove. I'll see if we can add the user id/name instead.

Reply:
done